AA-99: Adding in new date pills for the dates tab

Dillon-Dumesnil opened the pull request.

mikix commented on lms/djangoapps/courseware/courses.py:
```diff
@@ -462,30 +467,31 @@ def get_course_assignments(course_key, user, request, include_access=False):
-        assignments.append(_Assignment(block_key, title, url, date, requires_full_access))
+    # Ideally this function is always called with a request being passed in, but because it is also
+    # a subfunction of `get_course_date_blocks` which does not require a request, we are being defensive here.
+    if request:
```
Style nit: I generally prefer such early-exits to be explicit about what they are doing. Like, of the form:
```
if not request:
    return assignments  # or return []
```

Has the added bonus of keeping the "main path" less indented and thus it reads more like a main code path.

mikix commented on lms/djangoapps/courseware/views/views.py:
```diff
@@ -1078,7 +1078,7 @@ def dates(request, course_id):
     context = {
         'course': course,
-        'course_date_blocks': [block for block in course_date_blocks if block.title != 'current_datetime'],
+        'course_date_blocks': [block for block in course_date_blocks],
```
Could this just become `'course_date_blocks': course_date_blocks` or if you really just want to convert it to a list, `'course_date_blocks': list(course_date_blocks)`?

Dillon-Dumesnil commented:
Screenshot of the new pills 
<img width="1072" alt="image" src="https://user-images.githubusercontent.com/12799718/80532873-9f86f680-8951-11ea-9794-c3cdbbebd451.png">

Dillon-Dumesnil commented:
Audit Experience 
<img width="1072" alt="image" src="https://user-images.githubusercontent.com/12799718/80823746-0692ee00-8b92-11ea-9261-64f986a5864e.png">

Dillon-Dumesnil commented:
Updated Verified: 
<img width="1072" alt="image" src="https://user-images.githubusercontent.com/12799718/80831567-ed913980-8b9f-11ea-99ab-6248cfd24e13.png">

Updated Audit:
<img width="1072" alt="image" src="https://user-images.githubusercontent.com/12799718/80831602-fb46bf00-8b9f-11ea-8ea8-94a2830ec708.png">

Dillon-Dumesnil commented:
Updated since last screenshot: Line pixel width is 1px, not 2px

cpennington commented on lms/djangoapps/course_blocks/transformers/start_date.py:
```diff
@@ -71,7 +71,7 @@ def collect(cls, block_structure):
 
     def transform_block_filters(self, usage_info, block_structure):
         # Users with staff access bypass the Start Date check.
-        if usage_info.has_staff_access:
```
@ormsbee, @nasthagiri: Is it better to have a conditional like this, or to remove the transformer from the transformer stack for some uses? I'm not really sure how we wanted to handle conditional transforms like this.
